BedrockServer.cpp

```diff
@@ -1656,30 +1640,37 @@ void BedrockServer::_reply(unique_ptr<BedrockCommand>& command) {
     }
 }
 
-void BedrockServer::suppressCommandPort(const string& reason, bool suppress, bool manualOverride) {
-    // If we've set the manual override flag, then we'll only actually make this change if we've specified it again.
-    if (_suppressCommandPortManualOverride && !manualOverride) {
-        return;
+
+void BedrockServer::blockCommandPort(const string& reason) {
+    lock_guard<mutex> lock(_portMutex);
+    _commandPortBlockReasons.insert(reason);
+    if (_commandPortBlockReasons.size() == 1) {
+        _commandPortPublic = nullptr;
+        _portPluginMap.clear();
     }
-    SINFO((suppress ? "Suppressing" : "Clearing") << " command port due to: " << reason);
+    SINFO("Blocking command port due to: " <<  reason << (_commandPortBlockReasons.size() > 1 ? " (already blocked)" : "") << ".");
+}
 
-    // Save the state of manual override. Note that it's set to *suppress* on purpose.
-    if (manualOverride) {
-        _suppressCommandPortManualOverride = suppress;
+void BedrockServer::unblockCommandPort(const string& reason) {
+    lock_guard<mutex> lock(_portMutex);
+    _commandPortLikelyBlocked = true;
```

Comment:
I don't get this - instead of setting this when we block the command port, we set this when we attempt to unblock it. So if the command port is actually blocked, this will still be set to `false`. And if somebody tries to unblock it erroneously, we set this to `true` erroneously until the end of this function. 

Seems like this value is only useful if we block the port more than once then unblock it at least once. The rest of the time it doesn't seem all that helpful.

Reply:
Yep, this line is in the wrong method.